qt/scientific_interfaces/ISISReflectometry/GUI/Batch/BatchJobRunner.cpp

```diff
@@ -84,14 +85,55 @@ void BatchJobRunner::notifyReductionResumed() {
   if (m_rowLocationsToProcess.empty()) {
     // Nothing selected so process everything. Skip failed rows.
     m_processAll = true;
+    m_processPartial = false;
     m_reprocessFailed = false;
   } else {
     // User has manually selected items so only process the selection (unless
     // autoreducing). Also reprocess failed items.
     m_processAll = m_isAutoreducing;
     m_reprocessFailed = !m_isAutoreducing;
+    // Check whether a given group is in the selection. If not then check
```

Comment:
I think this whole block can probably be simplified to something like this:
```
    m_processPartial = false;
    if (!m_processAll) {
      // Check if a group will be partially processed, i.e. if it has some, but
      // not all, rows selected
      std::map<const int, size_t> selectedRowsPerGroup;
      for (auto location : m_rowLocationsToProcess) {
        if (isRowLocation(location))
          ++selectedRowsPerGroup[getParentGroupIndexFromPath(location)];
      }
      for (const auto groupIndexCountPair : selectedRowsPerGroup) {
        auto const groupIndex = groupIndexCountPair.first;
        auto const numSelected = static_cast<int>(groupIndexCountPair.second);
        m_processPartial = numSelected < getNumberOfInitialisedRowsInGroup(groupIndex);
        if (m_processPartial) {
          break;
        }
      }
    }
```

Reply:
@gemmaguest this mostly works, however if you select the group and one of its rows it will warn about partial processing.

Reply:
Ah, yes. How about something like this (if you haven't already got a solution):
```
    if (!m_processAll) {
      // Check if a group will be partially processed, i.e. if it has some, but
      // not all, rows selected
      std::map<const int, size_t> selectedRowsPerGroup;
      for (auto location : m_rowLocationsToProcess) {
        auto const groupIdx = groupOf(location);
        auto const totalRowsInGroup = getNumberOfInitialisedRowsInGroup(groupIdx);
        if (isGroupLocation(location))
          selectedRowsPerGroup[groupIdx] = totalRowsInGroup;
        else if (selectedRowsPerGroup[groupIdx] < totalRowsInGroup)
          ++selectedRowsPerGroup[groupIdx];
      }
      for (const auto groupIndexCountPair : selectedRowsPerGroup) {
        auto const groupIndex = groupIndexCountPair.first;
        auto const numSelected = static_cast<int>(groupIndexCountPair.second);
        m_processPartial = numSelected < getNumberOfInitialisedRowsInGroup(groupIndex);
        if (m_processPartial) {
          break;
        }
      }
    }
```